src/main/scala/scorex/core/network/DeliveryTracker.scala

```diff
@@ -82,22 +86,25 @@ class DeliveryTracker(system: ActorSystem,
     tryWithLogging {
       val checks = requested(modifierId).checks + 1
       setUnknown(modifierId)
-      if (checks < maxDeliveryChecks) setRequested(modifierId, modifierTypeId,  Some(cp), checks)
+      if (checks < maxDeliveryChecks) setRequested(modifierId, modifierTypeId, cp, checks)
```

Comment:
Shouldn't the check be `<=` instead of strict `<`? If the `maxDeliveryChecks` is 1, we increment the current checks counter 0 to 1 and we immediately skip the only one check. We could also increment the counter right after the if condition

Reply:
But then maxDeliveryChecks name would be misleading, cause we will actually do delivery check twice

Reply:
But if `maxDeliveryChecks` is set to 1 the first time we run this function the `checks` variable will be set to 1 and since 1 (`checks`) is not less than 1 (`maxDeliveryChecks`) a `StopExpectingError` will be raised and we checked none while the deliveryCheck was expected to be 1. It seems like we always check `maxDeliveryChecks` - 1 times

Reply:
The flow if next:
1. We add a modifier to the deliveryTracker, it is requested. At the same time, we create a scheduled CheckDelivery message. CheckDone = 0
2. 2 seconds elapsed, CheckDelivery message is not canceled, so it is sent.
3. NodeViewSynchronizer receives CheckDelivery, and checks(for the first time) if the modifier was received.
4. If not, it penalizes the peer for non-delivery, and gets into `onStillWaiting`, where it decides if it needs to re-schedule the check once again(second time)